### PR TITLE
Fix an issue with image size set to zero. Respects aspect ratio.

### LINKED ISSUE #3322: Img meta data - size doesn't work
image `Size` in "edit" modal does not set correct h/w in post. It appears to be overridden by the default/original upload size setting or by h/w=0.

- example where it gets overridden by default upload size (meta settings add "class="size-large" though meta says "large - 640 x 480"): https://cloudup.com/iekOCcC5HBd
- example where it is set to "small" in img settings > makes height & width = 0 in html until published > publish and it's overridden by default upload size: https://cloudup.com/clOwgPg9TKt

note: if themes set image size based on class (size-small, size-medium, and size-large), then let's remove dimensions or make them dynamic.

cc @aerych

### PR DESCRIPTION
**Pull request by aerych**
Closes #3322

### CODE REVIEW

**Comment by aerych** [comment]
cc @bummytime for a quick review

File: WordPress/Classes/ViewRelated/Post/EditImageDetailsViewController.m [hunk 1]
```diff
@@ -147,6 +151,17 @@ - (NSArray *)sizeTitles
     return _sizeTitles;
 }
 
+- (CGSize)sizeForSize:(CGSize)size constrainedToRatio:(CGFloat)ratio
```
Comment (bummytime) [inline WordPress/Classes/ViewRelated/Post/EditImageDetailsViewController.m]: @aerych I think this method is not quite working... portrait images are getting squished in wp-admin (landscape seem to be fine)

Comment (bummytime) [inline WordPress/Classes/ViewRelated/Post/EditImageDetailsViewController.m]: Ahh hold on...this may just be an issue with wp-admin

Comment (bummytime) [inline WordPress/Classes/ViewRelated/Post/EditImageDetailsViewController.m]: Yep, my bad.

**Comment by bummytime** [comment]
Tested. Full size fix seems to work well also. :shipit:

**Comment by bummytime** [comment]
Looks like @aerych found a portrait image issue...so I take back my ship it! :grinning:

**Comment by aerych** [comment]
@sendhil tested and gave his :shipit: in chat.

**Pull request merged**